csvjoin/functions.py

```diff
@@ -0,0 +1,257 @@
+        :type line1: str
+        :type line2: str
+        :type column_index1: int
+        :type column_index2: int
+    """
+    new_line = str(line1)
+    new_line = new_line.rstrip()
+    new_line = new_line.split(',')
+
+    new_line2 = str(line2)
+    new_line2 = new_line2.rstrip()
+    new_line2 = new_line2.split(',')
+
+    if column_index2 != -1:
+        new_line2.pop(column_index2)
+    else:
+        new_line.pop(column_index1)
+
+    new_line.extend(new_line2)
+
+    line_to_print = ''
+    for _ in range(len(new_line)):
+        line_to_print += '{: >13} '
+
+    print(line_to_print.format(*new_line))
+
+
+def split_and_strip(line: str) -> [str]:
+    """
+        Copies the string, deletes the \n sign and
+        splits by comma.
+
+        :param line: The line to split and rstrip.
+        :type line: str
+        :return: The array of strings.
+        :rtype: [str]
+   """
+    new_line = str(line)
+    new_line = new_line.rstrip()
+    return new_line.split(',')
+
+
+def print_non_inner_join(file, header: str, taken_rows: set, join_type: str, column_index: int):
+    """
+        Prints non-inner type of join in order to "fill" NULL values.
+
+        :param file: The handle of the file.
+        :param header: The header of the second file to get the
+        number of the NULLs.
+        :param taken_rows: The set of already printed rows of the file.
+        :param join_type: The type of the join.
+        :param column_index: The column_index to pass it
+        to the parse_data_line function.
+        :type file: TextIO
+        :type header: str
+        :type taken_rows: set
+        :type join_type: str
+        :type column_index: int
+    """
+    file.seek(0, 0)
+    file.readline()
+    line = file.readline()
+
+    # creating NULL line string
+    header = header.split(',')
+    n = len(header)
+    filler = ''
+    for _ in range(n - 1):
+        filler += 'NULL'
+        filler += ','
+    filler += 'NULL'
+
+    counter = 1
+    while line != '':
+        if counter in taken_rows:
+            line = file.readline()
+            counter += 1
+            continue
+
+        if join_type == 'left':
+            parse_data_line(line, filler, -1, column_index)
+
+        if join_type == 'right':
+            parse_data_line(filler, line, column_index, -1)
+
+        line = file.readline()
+        counter += 1
+
+
+def join_files(file_path1: str, file_path2: str, column_name: str, join_type: str):
+    """
+        The actual main algorithm for printing the joined files.
+
+        :param file_path1: The path of the first file.
+        :param file_path2: The path of the second file.
+        :param column_name: The name of the column to join on.
+        :param join_type: The type of the join.
+        :type file_path1: str
+        :type file_path2: str
+        :type column_name: str
+        :type join_type: str
+    """
+    try:
```

Comment:
1. Why can this occur? It's the first time I have seen this error.
2. Why the whole large function is wrapped in try/except? Can't this block be any smaller?